primefaces/src/main/resources/META-INF/resources/primefaces/datepicker/1-datepicker.js

```diff
@@ -107,11 +107,16 @@ PrimeFaces.widget.DatePicker = PrimeFaces.widget.BaseWidget.extend({
                     $this.jqEl.prop("readonly", true);
                 }
 
-                //user callback
+                // user callback
                 var preShow = $this.cfg.preShow;
                 if(preShow) {
                     return $this.cfg.preShow.call($this, inst);
                 }
+
+                // #7457 trigger view change if lazy model is used
+                if ($this.cfg.lazyModel) {
```

Comment:
I'm being conservative here. I think I'll drop this check and make the change global (consistent) and document it in the change log.

Reply:
@melloware what do you think? I'd like to remove these checks before merging the PR.

Reply:
Hmmm so always fire the event?  Let's default to conservative for right now.

Reply:
Alright. Although I do think that firing a change on open and not in close is better in general.

Let me change `event === null` to `!event` and I think we are good to go.

Reply:
Done